README.md

```diff
@@ -15,28 +15,129 @@ This is a promise based interface for interacting with a source control manageme
 ### configure
 The `configure` function takes in an object and resets the configuration values
 
-### formatScmUrl
+### parseUrl
 Required parameters:
 
 | Parameter        | Type  |  Description |
 | :-------------   | :---- | :-------------|
-| scmUrl        | String | Scm Url to format |
+| config             | Object | Configuration Object |
+| config.checkoutUrl | String | Checkout url for a repo to parse |
+| config.token  | String | The github token to check permissions on |
 
 #### Expected Outcome
-A formatted scm url to be used as a unique key. This function is synchronous.
+An scmUri (ex: `github.com:1234:branchName`, where 1234 is a repo ID number), which will be a unique identifier for the repo and branch in Screwdriver.
+
+#### Expected Promise response
+1. Resolve with an scm uri for the repository
+2. Reject if not able to parse url
+
+### parseHook
+Required parameters:
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| headers        | Object | Request header |
+| payload        | Object | Request payload |
+
+#### Expected Outcome
+An object with the following fields:
+```js
+{
+    type: 'pr',         // or 'repo'
+    action: 'opened',   // or 'closed' for type 'pr'; 'push' for type 'repo'
+    username: 'batman',
+    url: 'https://bitbucket.org/batman/test.git',
```

Comment:
Sorry for wrong link earlier, this should be `https://batman@bitbucket.org/batman/test.git`